deluge/core/torrent.py

```diff
@@ -1054,13 +1056,35 @@ def get_status(self, keys, diff=False, update=False, all_keys=False):
 
         return status_dict
 
-    def update_status(self, status):
+    def get_lt_status(self) -> 'lt.torrent_status':
+        """Get the torrent status fresh, not from cache.
+
+        This should be used when a guaranteed fresh status is needed rather than
+        `torrent.handle.status()` because it will update the cache as well.
+        """
+        self.status = self.handle.status()
+        return self.status
+
+    @property
+    def status(self) -> 'lt.torrent_status':
```

Comment:
It seems that using `lt.torrent_status` here and for above method is causing the autodoc errors. Using the full `libtorrent.torrent_status` works though :shrug:

Reply:
Actually using `lt` module seems to be the issue here so any other string would work

Reply:
I added some debug to sphinx_autodoc_typehints.process_docstring and I can see the issue is that it is encountering a `Mock object` which seems to be tripping it up. This is likely due to mocking libtorrent and other 3rd party modules in sphinx config.

Reply:
Fixed: 62a40521789eab27341